docs/migration/index.md

```diff
@@ -0,0 +1,382 @@
+    panic(err)
+  }
+  // Add the following:
+  app.UpgradeKeeper.SetModuleVersionMap(ctx, app.mm.GetVersionMap())
+  return app.mm.InitGenesis(ctx, app.appCodec, genesisState)
+}
+
+// Replace Marshaler with Codec
+func (app *App) AppCodec() codec.Codec {
+  return app.appCodec
+}
+
+// Replace BinaryMarshaler with BinaryCodec
+func initParamsKeeper(appCodec codec.BinaryCodec, legacyAmino *codec.LegacyAmino, key, tkey sdk.StoreKey) paramskeeper.Keeper {
+  //...
+}
+```
+
+### app/genesis.go
+
+```go
+// Replace codec.JSONMarshaler with codec.JSONCodec
+func NewDefaultGenesisState(cdc codec.JSONCodec) GenesisState {
+  //...
+}
+```
+
+### testutil/keeper/mars.go
+
+Add the following code:
+
+```go
+package keeper
+
+import (
+  "testing"
+
+  "github.com/cosmonaut/mars/x/mars/keeper"
+  "github.com/cosmonaut/mars/x/mars/types"
+  "github.com/cosmos/cosmos-sdk/codec"
+  codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+  "github.com/cosmos/cosmos-sdk/store"
+  storetypes "github.com/cosmos/cosmos-sdk/store/types"
+  sdk "github.com/cosmos/cosmos-sdk/types"
+  "github.com/stretchr/testify/require"
+  "github.com/tendermint/tendermint/libs/log"
+  tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+  tmdb "github.com/tendermint/tm-db"
+)
+
+func MarsKeeper(t testing.TB) (*keeper.Keeper, sdk.Context) {
+  storeKey := sdk.NewKVStoreKey(types.StoreKey)
+  memStoreKey := storetypes.NewMemoryStoreKey(types.MemStoreKey)
+
+  db := tmdb.NewMemDB()
+  stateStore := store.NewCommitMultiStore(db)
+  stateStore.MountStoreWithDB(storeKey, sdk.StoreTypeIAVL, db)
+  stateStore.MountStoreWithDB(memStoreKey, sdk.StoreTypeMemory, nil)
+  require.NoError(t, stateStore.LoadLatestVersion())
+
+  registry := codectypes.NewInterfaceRegistry()
+  k := keeper.NewKeeper(
+    codec.NewProtoCodec(registry),
+    storeKey,
+    memStoreKey,
+  )
+
+  ctx := sdk.NewContext(stateStore, tmproto.Header{}, false, log.NewNopLogger())
+  return k, ctx
+}
+```
+
+### testutil/network/network.go
+
+```go
+func DefaultConfig() network.Config {
+  //...
+  return network.Config{
+    //...
+    // Add sdk.DefaultPowerReduction
+    AccountTokens:   sdk.TokensFromConsensusPower(1000, sdk.DefaultPowerReduction),
+    StakingTokens:   sdk.TokensFromConsensusPower(500, sdk.DefaultPowerReduction),
+    BondedTokens:    sdk.TokensFromConsensusPower(100, sdk.DefaultPowerReduction),
+    //...
+  }
+}
+```
+
+### testutil/sample/sample.go
+
+Add the following code:
+
+```go
+package sample
+
+import (
+  "github.com/cosmos/cosmos-sdk/crypto/keys/ed25519"
+  sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// AccAddress returns a sample account address
+func AccAddress() string {
+  pk := ed25519.GenPrivKey().PubKey()
+  addr := pk.Address()
+  return sdk.AccAddress(addr).String()
+}
+```
+
+### go.mod
+
+```
+module github.com/cosmonaut/mars
+
+go 1.16
+
+require (
+	github.com/cosmos/cosmos-sdk v0.44.0
+	github.com/cosmos/ibc-go v1.2.0
+	github.com/gogo/protobuf v1.3.3
```

Comment:
if the user uses Bandchain oracles, he needs to update the package (`github.com/bandprotocol/bandchain-packet`) version to `v0.0.2`

Reply:
I've used the `go.mod` scaffolded by default with the latest `develop`. Should we also change the default `go.mod`?

Reply:
We only add the `go.mod` if we scaffold an oracle component, like the wasm integration. Maybe we should add a note: "if you are using Banchain Oracle integration, you should update to version `v0.0.2`"